### PR TITLE
Get email map and fix comment and auto assign

### PR DESCRIPTION
**Pull request by tapishr**
Please check the action items covered in the PR - 

- [x] Build is running
- [ ] Eventing is functional and tested
- [ ] Unit or integration tests added and running
- [x] Manual QA

Corresponding vibinex-server PR - [https://github.com/Alokit-Innovations/vibinex-server/pull/242](https://github.com/Alokit-Innovations/vibinex-server/pull/242)

This is what the comment looks like now. In this case, vibinex-server had gh login info for tapish303@gmail.com but not for other two aliases - 

image

### CODE REVIEW

File: vibi-dpu/src/bitbucket/repo.rs [hunk 1]
```diff
@@ -13,8 +13,9 @@ pub async fn get_workspace_repos(workspace: &str, access_token: &str) -> Option<
         let mut is_private = true;
         if is_private_res.is_none() {
             log::error!("[get_workspace_repos] Error in deserializing is_private_res: {:?}", &repo_json);
+        } else {
+            is_private = is_private_res.expect("Uncaught error in is_private_res");
```
Comment (tapishr) [inline vibi-dpu/src/bitbucket/repo.rs:17]: This resolves a warning, unrelated to current PR

File: vibi-dpu/src/core/github/setup.rs [hunk 2]
```diff
@@ -139,9 +139,8 @@ pub async fn setup_self_host_user_repos_github(access_token: &str) {
     let mut repo_owner_map: std::collections::HashMap<String, Vec<String>> = std::collections::HashMap::new();
 
     for repo in repos {
-        let token_copy = access_token.clone();
         let mut repo_copy = repo.clone();
-        clone_git_repo(&mut repo_copy, &token_copy, &repo_provider).await;
+        clone_git_repo(&mut repo_copy, access_token, &repo_provider).await;
```
Comment (tapishr) [inline vibi-dpu/src/core/github/setup.rs:143]: This resolves a warning, unrelated to current PR

File: vibi-dpu/src/core/coverage.rs [hunk 3]
```diff
@@ -36,25 +36,25 @@ pub async fn process_coverage(hunkmap: &HunkMap, review: &Review, repo_config: &
...
-        log::debug!("[add_github_reviewers] blame_author_opt = {:?}", &blame_author_opt);
-        if blame_author_opt.is_none() {
+    for (_, (_, provider_ids_opt)) in coverage_map.iter() {
+        if provider_ids_opt.is_none() {
             continue;
         }
-        let blame_author = blame_author_opt.expect("Empty blame_author_opt");
-        if reviewers.contains(&blame_author) || prhunk.author().to_string() == blame_author {
+        let provider_ids = provider_ids_opt.to_owned().expect("Empty provider_ids_opt");
+        let provider_id_opt = provider_ids.first();
```
Comment (MuskanPaliwal) [inline vibi-dpu/src/core/coverage.rs:52]: Assumption: We assumed that single email can have multiple github or bitbucket handles. So we just go for the first handle from the array of handles.

File: vibi-dpu/src/db/aliases.rs [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+    }
+    let alias: Vec<String> = alias_res.expect("Uncaught error in alias_res");
+    return Some(alias);
+}
+
+pub fn update_handles_in_db(alias_key: &str, provider: &str, new_logins: Vec<String>) -> Vec<String>{
+    // Retrieve existing aliases from the database
+    let existing_aliases = match get_handles_from_db(alias_key, provider) {
+        Some(aliases) => aliases,
+        None => Vec::new(), // If no aliases found, initialize as empty vector
```
Comment (MuskanPaliwal) [inline vibi-dpu/src/db/aliases.rs:70]: why are we using match here?

File: vibi-dpu/src/utils/aliases.rs [hunk 5]
```diff
@@ -0,0 +1,27 @@
+use crate::{core::utils::get_aliases, db::aliases::get_handles_from_db};
+
+use super::review::Review;
+
+
+pub async fn get_login_handles(git_alias: &str, review: &Review) -> Option<Vec<String>> {
+    // Get aliases from db
+    let handles_db_opt = get_handles_from_db(git_alias, &review.provider());
+    if handles_db_opt.is_none() {
+        let server_aliases_opt = get_aliases(review).await;
```
Comment (MuskanPaliwal) [inline vibi-dpu/src/utils/aliases.rs:10]: can we please have a better name here for the function? `get_handles_from_server`?

File: vibi-dpu/src/utils/aliases.rs [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+        log::debug!("[get_login_handles] server_aliases_opt = {:?}", &server_aliases_opt);
+        if server_aliases_opt.is_none() {
+            log::debug!("[get_login_strings] No login strings found for git alias {} from server or db", git_alias);
+            return None;
+        }
+        let server_alias_map = server_aliases_opt.expect("Empty server_aliases_opt");
+        let login_handles_opt = server_alias_map.get(git_alias);
+        if login_handles_opt.is_none() {
+            return None;
```
Comment (MuskanPaliwal) [inline vibi-dpu/src/utils/aliases.rs:19]: Add logs for both the return statements

**Review by MuskanPaliwal — CHANGES_REQUESTED** [review]

File: vibi-dpu/src/utils/aliases.rs (hunk 6 above)
Comment (tapishr) [reply vibi-dpu/src/utils/aliases.rs:21]: done

File: vibi-dpu/src/utils/aliases.rs (hunk 5 above)
Comment (tapishr) [reply vibi-dpu/src/utils/aliases.rs]: done

File: vibi-dpu/src/db/aliases.rs (hunk 4 above)
Comment (tapishr) [reply vibi-dpu/src/db/aliases.rs]: removed match

**Review by MuskanPaliwal — APPROVED** [review]

**Review by MuskanPaliwal — APPROVED** [review]

**Pull request merged**